### PR TITLE
Lots of little docs cleanup after one pass

### PR DESCRIPTION
**Pull request by edwardleardi**
Consistent periods
Consistent sphinx semantic role usage
Typos
Missing docstrings

---

## Checklist:

<details open>

<summary></summary>

- Does this pull request close an issue? We encourage you to open an issue first if this pull request (PR) is not a
  minor change.
  - [x] No
    - [x] The change in this pull request is minor.
  - [ ] Yes
    - Close #{issue_number}

For the following questions, only check the boxes that are applicable.

- [x] Ensure that the pull request text above the horizontal line is descriptive.
- [x] Add/update relevant tests.
- [x] Add/update relevant documents.

- Do you have triage permission of this repository?
  - [ ] No
    - You are good.
  - [x] Yes
    - Does this pull request close an issue?
      - [x] No, this pull request also acts as an issue by itself.
        - [x] Assign yourself.
        - [x] Label this pull request.
        - [x] Put a milestone.
        - [x] Put this pull request under "In Progress" or "Under Review" pipeline on ZenHub.
        - [x] Put this pull request under the appropriate epic on ZenHub.
        - [x] Put an estimate on ZenHub.
        - [x] Add dependency relationship with other issues/pull requests on ZenHub.
      - [ ] Yes, this pull request addresses an issue and does not act as an issue.
        - [ ] Connect this pull request to the underlying issue on ZenHub.
        - [ ] DON'T:
          - DON'T assign yourself or anyone else.
          - DON'T label.
          - DON'T put a milestone.
          - DON'T put this pull request under any epic on ZenHub.
          - DON'T put an estimate on ZenHub.
    - [x] When the pull request is ready, request review.
</details>

### CODE REVIEW

File: pydax/loaders/_base.py [hunk 1]
```diff
@@ -40,7 +44,7 @@ def check_path(self, path: Union[_typing.PathLike, Dict[str, str]]) -> None:
 
         :param path: The path of the file to be loaded.
         :raises TypeError: ``path`` is not a path object.
-        :return: No return unless the path is invalid, in which case see TypeError.
+        :return: No return unless the ``path`` is invalid, in which case see ``TypeError``.
```
Comment (xuhdev) [inline pydax/loaders/_base.py]:
```suggestion
        :return: No return unless the ``path`` is invalid, in which case see :class:`TypeError`.
```

We can set up interlink with the docs of python and pandas: https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html

File: pydax/_high_level.py [hunk 2]
```diff
@@ -64,7 +65,7 @@ def init(update_only: bool = True, **kwargs: Any) -> None:
     :param LICENSE_SCHEMA_URL: The default license schema file URL.
     :param DATADIR: Default dataset directory to download/load to/from. The path can be either absolute or relative to
         the current working directory, but will be converted to the absolute path immediately in this function.
-        Defaults to: ~/.pydax/data
+        Defaults to: :file:`~/.pydax/data`.
```
Comment (xuhdev) [inline pydax/_high_level.py:68]: Another issue is what this means for Windows users. We can fix later if you what to.

File: pydax/loaders/_base.py (hunk 1 above)
Comment (edwardleardi) [reply pydax/loaders/_base.py]: nice I didn't know about that extension, I'll add it

File: pydax/_high_level.py (hunk 2 above)
Comment (edwardleardi) [reply pydax/_high_level.py:68]: yeah agree we can fix later

File: pydax/loaders/_table.py [hunk 3]
```diff
@@ -26,8 +27,10 @@
 
 
 class CSVPandasLoader(Loader):
+    """CSV to Pandas dataframe loader.
+    """
 
-    def load(self, path: Union[_typing.PathLike, Dict[str, str]], options: SchemaDict) -> str:
+    def load(self, path: Union[_typing.PathLike, Dict[str, str]], options: SchemaDict) -> pd.DataFrame:
```
Comment (xuhdev) [inline pydax/loaders/_table.py:33]: I wonder why this wasn't caught by mypy

**Review by xuhdev — APPROVED** [review]

File: pydax/loaders/_table.py (hunk 3 above)
Comment (edwardleardi) [reply pydax/loaders/_table.py:33]: yeah I was surprised too when I noticed it

**Pull request merged**